rules/library/xcconfig.bzl

```diff
@@ -338,3 +325,27 @@ def copts_by_build_setting_with_defaults(xcconfig = {}, fetch_default_xcconfig =
         ibtool_copts = select(ibtool_copts_by_build_setting),
         linkopts = select(linkopts_by_build_setting),
     )
+
+def merge_xcconfigs(*xcconfigs):
+    """Merges a list of xcconfigs into a single dictionary
+
+    Uses some heuristics to merge xcconfigs in a way that is compatible with Xcode's behavior:
+
+    - If a key is present in multiple xcconfigs
+        - If the value is a list, the values are concatenated
+        - If the value is a string, the last xcconfig value is used
+
+    Args:
+        *xcconfigs: A list of dictionaries of Xcode build settings
+    Returns:
+        A dictionary of Xcode build settings
+    """
+    merged_xcconfig = {}
+    for xcconfig in xcconfigs:
+        for (key, value) in dict(xcconfig).items():
+            if types.is_list(merged_xcconfig.get(key, None)) and types.is_list(value):
+                merged_xcconfig[key] = merged_xcconfig[key] + value
```

Comment:
Can this contain duplicates? Should we remove if so?

Reply:
Id be inclined to not remove duplicates, that feels like user error we shouldn't correct for but happy to reconsider if folks feel different!

Reply:
Up to you! Agree that this would most likely apply to user error, so makes sense to me 👍